### PR TITLE
Three-dimensional detrending

### PR DESCRIPTION
**Pull request by roxyboy**
I added the option to apply detrending over 3 dimensions simultaneously, instead of just a 2D plane.

### CODE REVIEW

**Comment by roxyboy** [comment]
I also refactored to windowing function a bit.

File: xrft/xrft.py [hunk 1]
```diff
@@ -196,12 +196,9 @@ def dft(da, dim=None, shift=True, detrend=None, window=False):
                 raise ValueError("Data should be dask array.")
 
     if window:
-        window_array = _create_window(da, dim)
-        da_win = da * window_array
-    else:
-        da_win = da
+        da = _apply_window(da, dim)
```
Comment (roxyboy) [inline xrft/xrft.py]: I refactored the windowing function a bit.

File: xrft/xrft.py [hunk 2]
```diff
@@ -10,7 +10,7 @@
 from functools import reduce
 import operator
 
-__all__ = ["detrend2","_detrend_wrap",
+__all__ = ["_detrend","_detrend_wrap",
```
Comment (rabernat) [inline xrft/xrft.py]: the underscore indicates a private function. Such functions shouldn't be part of the public API

File: xrft/xrft.py [hunk 3]
```diff
@@ -97,25 +122,42 @@ def func(a, axes=None):
...
+def _apply_detrend(da, axis_num):
+    """Wrapper function for applying detrending"""
+    if da.chunks:
+        func = _detrend_wrap(_detrend)
+        da = xr.DataArray(func(da.data, axes=axis_num),
+                        dims=da.dims, coords=da.coords)
+    else:
+        if da.ndim == 1:
+            da = xr.DataArray(sps.detrend(da),
+                            dims=da.dims, coords=da.coords)
```
Comment (rabernat) [inline xrft/xrft.py]: So you use scipy for 1D detrending and your own custom function for higher dimensions?

Comment (roxyboy) [inline xrft/xrft.py]: I ended up doing that because the scipy function doesn't take data with more than one dimension.

**Pull request merged**